analysis/study_definition.py

```diff
@@ -292,6 +402,39 @@
             "incidence": 0.95,
         },
     ),
+    #### Subcategorise recent hba1c measures in no recent measure (0); measure indicating controlled diabetes (1);
+    #### measure indicating uncontrolled diabetes (2)
+    hba1c_category = patients.categorised_as(
```

Comment:
You may also want to return the operator attached to these numeric values as these can often be recorded in unexpected ways. e.g If the returned `hba1c_mmol_per_mol` is `>20`, you probably want to set that to NULL as it doesn't fall nicely into one of your categories. [This is an example of this for egfr](https://github.com/opensafely/pincer-measures/blob/main/analysis/study_definition.py#L410). @robinyjpark may be able to tell you if there are any oddities with hba1c numeric values

Reply:
No worries! I've copied this from earlier studies studying subgroups at risk. Not sure what's best here? @CarolineMorton?

Reply:
48 and 58 have been used clinically for different reasons so it gets a bit confusing. Technically, 48 is both the cut off for diagnosis of diabetes and the target for control in diabetic patients. However, 58 is used as the threshold for intensifying drug therapy in diabetics and 48 - 58 might be a bit more a "wait and see" approach. See [ref](https://www.nice.org.uk/guidance/ng28/chapter/Recommendations#hba1c-measurement-and-targets)

Reply:
Thx! I'll stick to 58.